File: uta-repo/snippets/ruby/verify.rb
#!/usr/bin/env ruby
# UTA — Ruby example: verify a credential via public HTTP API.
# Repo: https://github.com/alicelabs-llc/universal-trust-adapter
require 'net/http'
require 'uri'
require 'json'

card = ARGV[0] || abort('Usage: verify.rb <card-string>')
uri = URI('https://www.marketnow.site/api/trust?action=verify')
req = Net::HTTP::Post.new(uri, 'Content-Type' => 'application/json')
req.body = { card: card }.to_json

resp = Net::HTTP.start(uri.host, uri.port, use_ssl: true) { |http| http.request(req) }
result = JSON.parse(resp.body)
puts "Decision:   #{result['decision']}"
puts "Format:     #{result['detected_format']}"
puts "Issuer:     #{result['issuer']}"
exit case result['decision']
     when 'PERMIT' then 0
     when 'DENY' then 1
     else 2
     end

File: uta-repo/snippets/php/verify.php
<?php
/**
 * UTA — PHP example: verify a credential via public HTTP API.
 * Repo: https://github.com/alicelabs-llc/universal-trust-adapter
 */
$card = $argv[1] ?? die("Usage: php verify.php <card-string>\n");
$ch = curl_init('https://www.marketnow.site/api/trust?action=verify');
curl_setopt_array($ch, [
    CURLOPT_POST => true,
    CURLOPT_HTTPHEADER => ['Content-Type: application/json'],
    CURLOPT_POSTFIELDS => json_encode(['card' => $card]),
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_TIMEOUT => 10,
]);
$response = curl_exec($ch);
curl_close($ch);
$result = json_decode($response, true);
echo "Decision:   {$result['decision']}\n";
echo "Format:     {$result['detected_format']}\n";
echo "Issuer:     {$result['issuer']}\n";
exit(match($result['decision']) {
    'PERMIT' => 0, 'DENY' => 1, default => 2,
});
